tensorflow/compiler/tf2tensorrt/convert/ops/layer_utils.h

```diff
@@ -0,0 +1,365 @@
+    TRT_ENSURE(layer);
+    return layer;
+  }
+
+  // Adds an Equal operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> Equal(
+      nvinfer1::ITensor* lhs, nvinfer1::ITensor* rhs) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(lhs);
+    TRT_EXPECT(rhs);
+    nvinfer1::IElementWiseLayer* layer = network_->addElementWise(
+        *lhs, *rhs, nvinfer1::ElementWiseOperation::kEQUAL);
+    TRT_ENSURE(layer);
+    return layer;
+  }
+
+  // Simulates the creation of a tensor of the same shape as "input", but filled
+  // with 0's where "input" is non-zero. TensorRT doesn't have dynamic creation
+  // functions, so this function multiplies the input by a zero value to side
+  // step knowing the shape. For static inputs, this can be optimized using one
+  // of the TRTNetworkBuilder::Constant function instead.
+  StatusOr<nvinfer1::IElementWiseLayer*> ZerosLikeDyamic(
+      nvinfer1::ITensor* input) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(input);
+    StatusOr<nvinfer1::IConstantLayer*> zero =
+        this->Constant<int32>(0, input->getDimensions().nbDims);
+    TRT_ENSURE_OK(zero);
+    return Mul(input, (*zero)->getOutput(0));
+  }
+
+  // Adds an Add operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> FloorDiv(
+      nvinfer1::ITensor* lhs, nvinfer1::ITensor* rhs) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(lhs);
+    TRT_EXPECT(rhs);
+    nvinfer1::IElementWiseLayer* layer = network_->addElementWise(
+        *lhs, *rhs, nvinfer1::ElementWiseOperation::kFLOOR_DIV);
+    TRT_ENSURE(layer);
+    return layer;
+  }
+
+  // Abs(Ceiling divide(x/y)) operation for integers
+  StatusOr<nvinfer1::IElementWiseLayer*> AbsCeilDivInt(
+      nvinfer1::ITensor* lhs, nvinfer1::ITensor* rhs) noexcept {
+    TRT_EXPECT(network_);
+    TRT_EXPECT(lhs);
+    TRT_EXPECT(rhs);
+    StatusOr<nvinfer1::IElementWiseLayer*> rhs_abs = this->AbsInt(rhs);
+    TRT_ENSURE_PTR_OK(rhs_abs);
+    StatusOr<nvinfer1::IElementWiseLayer*> lhs_abs = this->AbsInt(lhs);
+    TRT_ENSURE_PTR_OK(lhs_abs);
+    StatusOr<nvinfer1::IElementWiseLayer*> add1 =
+        this->Add((*lhs_abs)->getOutput(0), (*rhs_abs)->getOutput(0));
+    TRT_ENSURE_PTR_OK(add1);
+    StatusOr<nvinfer1::IConstantLayer*> one_const =
+        this->Constant<int32>(1, rhs->getDimensions().nbDims);
+    TRT_ENSURE_PTR_OK(one_const);
+    StatusOr<nvinfer1::IElementWiseLayer*> numerator =
+        this->Sub((*add1)->getOutput(0), (*one_const)->getOutput(0));
+    TRT_ENSURE_PTR_OK(numerator);
+    return FloorDiv((*numerator)->getOutput(0), (*rhs_abs)->getOutput(0));
+  }
+
+  // Adds an operation whose output is equal to "true_values" where the
+  // corresponding value in the "bool_condition" tensor is true, otherwise the
+  // output has the corresponding value from "false_values". In TensorRT this is
+  // called "ISelectLayer", but to conform with Tensorflow naming, we use
+  // "where" here.
```

Comment:
see https://www.tensorflow.org/api_docs/python/tf/where
tf.where also supports the case where X and Y are not provide, in such as case, it returns the indices in the condition with true values. this is probably the reason that it is called where. Does ISeleceLayer also allow the two input to be null and return the indices in the condition? If not, we should just call this routine Select.

Reply:
I just removed this operation to avoid use of ISelectLayer